app/src/main/java/com/yacgroup/yacguide/list_adapters/BaseViewAdapter.kt

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (C) 2022 Axel Paetzold
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package com.yacgroup.yacguide.list_adapters
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageButton
+import android.widget.LinearLayout
+import android.widget.TextView
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import com.yacgroup.yacguide.R
+
+class BaseViewAdapter(
+    private val _onClick: (Int) -> Unit,
+    private val _button1IconResource: Int,
```

Comment:
Note: I added the button icons here and not statically in the xml because I want to re-use this class for the RegionManager view